src/Model.php

```diff
@@ -0,0 +1,582 @@
+	 * @param  array  $options
+	 * @return bool
+	 */
+	 public function update(array $attributes = [], array $options = [])
+	 {
+		 if (!$this->exists) {
+			 return false;
+		 }
+
+		 // TODO is it a lot better performance wise if we only save the modified attributes?
+		 return $this->setAttributes($attributes)->save($options);
+	 }
+
+	/**
+	 * Delete the model from the database.
+	 *
+	 * @return bool|null
+	 *
+	 * @throws \Exception
+	 */
+	public function delete()
+	{
+		if (is_null($this->getPrimaryKey())) {
+			throw new Exception('No primary key defined on model.');
+		}
+
+		if (!$this->exists) {
+			return false;
+		}
+
+		//TODO relations to be taken cared of here.
+
+		$query = $this->newQuery();
+
+		return $this->performDelete($query);
+	}
+
+	/**
+	 * Save the model to the database.
+	 *
+	 * @param  array  $options
+	 * @return bool
+	 */
+	public function save(array $options = [])
+	{
+		$query = $this->newQuery();
+
+		// If the model already exists in the database we can just update our record
+		// that is already in this database using the current IDs in this "where"
+		// clause to only update this model. Otherwise, we'll just insert them.
+		if ($this->exists) {
+			$saved = $this->isDirty() ?
+				$this->performUpdate($query) : true;
+		}
+
+		// If the model is brand new, we'll insert it into our database and set the
+		// ID attribute on the model to the value of the newly inserted row's ID
+		// which is typically an auto-increment value managed by the database.
+		else {
+			$saved = $this->performInsert($query);
+		}
+
+		// If the model is successfully saved, we need to do a few more things once
+		// that is done. We will call the "saved" method here to run any actions
+		// we need to happen after a model gets successfully saved right here.
+		if ($saved) {
+			$this->syncOriginal();
+		}
+
+		return $saved;
+	}
+
+	/**
+	 * Perform a model insert operation.
+	 *
+	 * @param  Query  $query
+	 * @return bool
+	 */
+	 protected function performInsert(Query $query)
+	 {
+		 if (empty($this->attributes)) {
+			 return true;
+		 }
+
+		 $success = $query->insert();
+
+		 if ($success){
+			 $this->exists = true;
+		 }
+
+		 return $success;
+	 }
+
+	/**
+	 * Perform a model insert operation.
+	 *
+	 * @param  Query  $query
+	 * @return bool
+	 */
+	 protected function performUpdate($query)
+	 {
+		 if (empty($this->attributes)) {
+			 return true;
+		 }
+
+		 $success = $query->update();
+
+		 return $success;
+	 }
+
+	/**
+	 * Perform a model insert operation.
+	 *
+	 * @param  Query  $query
+	 * @return bool
+	 */
+	protected function performDelete($query)
+	{
+		$success = $query->delete($this);
+
+		if ($success)
+		{
+			$this->exists = false;
+		}
+
+		return $success;
+	}
+
+
+	/**
+	 * Get a new query builder for the model's table.
+	 *
+	 * @return Query
+	 */
+	public function newQuery()
+	{
+		return new Query($this->_dbFactory->getQuery($this->_db->getName(), $this->_db), $this->_db, $this);
```

Comment:
We should be able to call `$this->_db->getQuery(true)` here instead of going through the db factory (see https://github.com/joomla-framework/database/blob/2.0-dev/src/DatabaseDriver.php#L1021 then we also don't need to have the `$this->_dbFactory` stored in the class)